build.sbt

```diff
@@ -154,6 +154,11 @@ val mimaFilterSettings = Seq {
     ProblemFilters.exclude[ReversedMissingMethodProblem]("scala.reflect.runtime.SynchronizedTypes.scala$reflect$runtime$SynchronizedTypes$$super$defineNormalized"),
     ProblemFilters.exclude[DirectMissingMethodProblem]("scala.reflect.runtime.SynchronizedTypes.defineNormalized"),
     ProblemFilters.exclude[DirectMissingMethodProblem]("scala.reflect.runtime.JavaUniverse.defineNormalized"),
+    ProblemFilters.exclude[MissingClassProblem]("scala.collection.immutable.Set$SetNIterator"),
+    ProblemFilters.exclude[IncompatibleResultTypeProblem]("scala.collection.immutable.Set#Set3.filter"),
+    ProblemFilters.exclude[IncompatibleResultTypeProblem]("scala.collection.immutable.Set#Set4.filter"),
+    ProblemFilters.exclude[IncompatibleResultTypeProblem]("scala.collection.immutable.Set#Set2.filter"),
+    ProblemFilters.exclude[IncompatibleResultTypeProblem]("scala.collection.immutable.Set#Set1.filter"),
```

Comment:
This is a bit questionable. `SetN` are not private types, but their constructors are `private[collection]`.

```scala
object Test extends App {
  collection.immutable.Set("a", "b") match {
   case set2: collection.immutable.Set.Set2[String] => set2.filter(_ => true)
  }
}
```
```
$ scala -version
Scala code runner version 2.13.1 -- Copyright 2002-2019, LAMP/EPFL and Lightbend, Inc.
$ qscala -version
Scala code runner version 2.13.2-20191024-071344-7a0c276 -- Copyright 2002-2019, LAMP/EPFL and Lightbend, Inc.

$ qscalac sandbox/test.scala && qscala Test
$ scala Test
Exception in thread "main" java.lang.NoSuchMethodError: scala.collection.immutable.Set$Set2.filter(Lscala/Function1;)Lscala/collection/immutable/Set;
	at Test$.delayedEndpoint$Test$1(test.scala:3)
	at Test$delayedInit$body.apply(test.scala:1)
```

That's pretty questionable user code, we could choose to break it anyway. But are there more reasonable ways that user code with 2.13.2 linking against the new `SetN.filter` methods?

Reply:
These types shouldn't appear in any signatures but your example is legitimate Scala code. I don't think we should break it. There are no hacks via Java or protected packages needed to make code break from such a change.

Reply:
I realised that I can (and should) be overriding `filterImpl`. This is `protected[collection]` so should not be linked to from client Scala code.